File: Binary Search/C++/loop.cpp
#include <iostream>
#include <ctype.h>

int binarySearch(int *array, size_t array_size, int target);

int main() {
    size_t array_size = 100000;
    int array[array_size]{}, target = 1;

    for (size_t i = 0; i < array_size; i++) {
        array[i] = i+1;
    }

    int output = binarySearch(&array[0], array_size, target);

    std::cout << "Element " << target << " found at " << output << " index." << std::endl;
    return 0;
}

int binarySearch(int *array, size_t array_size, int target) {
    uint32_t left = 0, right = array_size;

    while (right >= left) {
        uint32_t mid = left + (right - left) / 2;
        int current = array[mid];

        if (current == target) {
            return mid;
        }
        if (current < target) {
            left = mid;
        } else {
            right = mid;
        }
    }

    return -1;
}

File: Binary Search/Python/loop.py
def binary_search(array, target):
    left = 0
    right = len(array) - 1

    while right >= left:
        mid = left + (right - left) // 2
        current = array[mid]

        if current == target:
            return mid
        if current <= target:
            left = mid + 1
        else:
            right = mid - 1

    return -1


if __name__ == '__main__':
    array_length = 1000000
    target = 1
    array = [x + 1 for x in range(array_length)]

    output = binary_search(array, target)
    print(f'Element {target} found at {output} index.')
